api/src/main/java/org/pmiops/workbench/api/AppController.java

```diff
@@ -1,17 +1,63 @@
 package org.pmiops.workbench.api;
 
+import javax.inject.Provider;
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.pmiops.workbench.db.dao.WorkspaceDao;
+import org.pmiops.workbench.db.model.DbUser;
+import org.pmiops.workbench.db.model.DbWorkspace;
+import org.pmiops.workbench.exceptions.NotFoundException;
+import org.pmiops.workbench.leonardo.LeonardoApiClient;
+import org.pmiops.workbench.leonardo.LeonardoApiHelper;
 import org.pmiops.workbench.model.App;
 import org.pmiops.workbench.model.EmptyResponse;
 import org.pmiops.workbench.model.ListAppsResponse;
+import org.pmiops.workbench.model.WorkspaceAccessLevel;
+import org.pmiops.workbench.workspaces.WorkspaceAuthService;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.RestController;
 
 @RestController
 public class AppController implements AppApiDelegate {
+  private final LeonardoApiClient leonardoApiClient;
+  private final Provider<DbUser> userProvider;
+  private final WorkspaceAuthService workspaceAuthService;
+  private final Provider<WorkbenchConfig> workbenchConfigProvider;
+  private final WorkspaceDao workspaceDao;
```

Comment:
I generally prefer to go to the service layer from the controller instead of going to the dao directly

Reply:
+1 I copied this from RuntimeContrller. As RW codebase envolves overtime, it's getting bit messy between Controller vs Service vs Dao, including unittest....